elide-contrib/elide-test-helpers/src/main/java/com/yahoo/elide/contrib/testhelpers/graphql/elements/ValueWithVariable.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.contrib.testhelpers.graphql.elements;
+
+import java.io.Serializable;
+
+/**
+ * {@link ValueWithVariable} models the {@code valueWithVariable} defined in GraphQL grammar.
+ * <p>
+ * GraphQL grammar(6.0) for {@code valueWithVariable} consists of sub-definitions:
+ * <pre>
+ * {@code
+ *     valueWithVariable :
+ *         variable |
+ *         IntValue |
+ *         FloatValue |
+ *         StringValue |
+ *         BooleanValue |
+ *         NullValue |
+ *         enumValue |
+ *         arrayValueWithVariable |
+ *         objectValueWithVariable;
+ * }
+ * </pre>
+ * Each sub-definition is an implementation of {@link ValueWithVariable}.
+ * <p>
+ * This is a {@link java.util.function functional interface} whose functional method is {@link #toGraphQLSpec()}.
+ */
+@FunctionalInterface
+public interface ValueWithVariable extends Serializable {
```

Comment:
I'd like to see if it's possible remove this interface and all of its implementations and just use the java objects directly (a string, an integer, an actual enum, an object) in place of these.

Reply:
Done